### PR TITLE
FINERACT-1765: Introduce Lombok to gcm module

### PR DESCRIPTION
**Pull request by hmg65**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/domain/DeviceRegistration.java [hunk 1]
```diff
@@ -42,40 +49,8 @@ public final class DeviceRegistration extends AbstractPersistableCustom {
...
-        this.registrationId = registrationId;
-    }
-
-    public LocalDateTime getUpdatedOnDate() {
-        return this.updatedOnDate;
-    }
-
-    public void setUpdatedOnDate(LocalDateTime updatedOnDate) {
-        this.updatedOnDate = updatedOnDate;
+        return new DeviceRegistration().setRegistrationId(registrationId).setClient(client);
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/domain/DeviceRegistration.java]: `this.updatedOnDate = DateUtils.getLocalDateTimeOfTenant(); ` -> this is missing according to the original constructor.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/domain/MulticastResult.java [hunk 2]
```diff
@@ -83,64 +92,14 @@ private MulticastResult(Builder builder) {
...
-
-    /**
-     * Gets additional ids if more than one multicast message was sent.
-     */
-    public List<Long> getRetryMulticastIds() {
-        return retryMulticastIds;
-    }
-
     @Override
     public String toString() {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/domain/MulticastResult.java]: You can replace the toString with a @ToString lombok annotation.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/domain/DeviceRegistration.java (hunk 1 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/domain/DeviceRegistration.java]: done

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/domain/MulticastResult.java (hunk 2 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/gcm/domain/MulticastResult.java]: @toString is already included in @Data annotation

**Review by galovics — APPROVED** [review]

**Pull request merged**